pkg/exchange/coinbase/exchange.go

```diff
@@ -539,41 +539,56 @@ func (e *Exchange) queryProductTradesByPagination(
 	req.ProductID(toLocalSymbol(symbol))
 	if options.StartTime != nil {
 		req.StartDate(
-			options.StartTime.Format(time.RFC3339Nano),
+			options.StartTime.Format("2006-01-02"),
 		)
 	}
 	if options.EndTime != nil {
+		// end_date is exclusive, add one day to the end date
 		req.EndDate(
-			options.EndTime.Format(time.RFC3339Nano),
+			options.EndTime.AddDate(0, 0, 1).Format("2006-01-02"),
 		)
 	}
-	if options.LastTradeID > 0 {
-		req.Before(options.LastTradeID)
-	}
-	cbTrades, err = req.Do(ctx)
+	cbTradesDirty, err := req.Do(ctx)
 	if err != nil {
 		return nil, err
 	}
-
+	seenTrades := make(map[uint64]struct{})
+	for _, trade := range cbTradesDirty {
+		seenTrades[trade.TradeID] = struct{}{}
+	}
```

Comment:
not sure why do we need this?

Reply:
Ya, you're right.
It's my experiment codes which I forgot to delete it.
Will fix it.